### DISCUSSION TITLE
blog/organizing-system-configs-with-nixos

### DISCUSSION THREAD

**Discussion by giscus** [JRMurr/JRMurr.github.io#9] [Comments]
# blog/organizing-system-configs-with-nixos

How I organize and manage my system and user configs with NixOS and homemanager

https://johns.codes/blog/organizing-system-configs-with-nixos

**Comment by SuperSandro2000** [comment]
If you are exclusive on NixOS then I wouldn't recommend home-manager. It lacks the value compared to a normal dotfile manager for the added complexity. Often you just proxy the config through it and don't get high value settings like complete and complex nginx location blocks. That's mostly due to them having a correct answer while dotfiles don't have that.

**Reply by JRMurr** [reply]
That's a fair criticism.

There are a fair number of home manager options that are pretty basic and I have many things configured with just symlinking the config file i manage.

The main benefit for me is having "1 tool" that does all config so I'm happy with the tradeoff.


For complex things like nginx I would try to use "offical" nixos options if I can

**Reply by kerrnelPanic** [reply]
I get the criticism about hm adding another layer and thereby cutting off some features, but I don't quite get why you wouldn't use hm because you're on NixOS only. What's the point here? Hm will cut features on other platforms as well.
I try to use it mainly as a package manager per user, so it only manages the apps available for a user and the apps' config stays in dotfiles or where ever the app has it.

**Reply by SuperSandro2000** [reply]
I can add per user packages with `users.users.<name>.packages`.

In my setup with dotfiles, hm just does not have its place as it doesn't do anything I need.
